Restic issues

I will amend when/if I get more errors.

* MacOS + clang (upstream Travis CI).
```
poly1305-donna.c:18:6: error: macro expansion producing 'defined' has undefined behavior [-Werror,-Wexpansion-to-defined]
#if (HAS_SIZEOF_INT128_64BIT || HAS_MSVC_64BIT || HAS_GCC_4_4_64BIT)
     ^
poly1305-donna.c:14:34: note: expanded from macro 'HAS_SIZEOF_INT128_64BIT'
#define HAS_SIZEOF_INT128_64BIT (defined(__SIZEOF_INT128__) && defined(__LP64__))
                                 ^
poly1305-donna.c:18:6: error: macro expansion producing 'defined' has undefined behavior [-Werror,-Wexpansion-to-defined]
poly1305-donna.c:14:64: note: expanded from macro 'HAS_SIZEOF_INT128_64BIT'
#define HAS_SIZEOF_INT128_64BIT (defined(__SIZEOF_INT128__) && defined(__LP64__))
                                                               ^
poly1305-donna.c:18:33: error: macro expansion producing 'defined' has undefined behavior [-Werror,-Wexpansion-to-defined]
#if (HAS_SIZEOF_INT128_64BIT || HAS_MSVC_64BIT || HAS_GCC_4_4_64BIT)
                                ^
poly1305-donna.c:15:25: note: expanded from macro 'HAS_MSVC_64BIT'
#define HAS_MSVC_64BIT (defined(_MSC_VER) && defined(_M_X64))
                        ^
poly1305-donna.c:18:33: error: macro expansion producing 'defined' has undefined behavior [-Werror,-Wexpansion-to-defined]
poly1305-donna.c:15:46: note: expanded from macro 'HAS_MSVC_64BIT'
#define HAS_MSVC_64BIT (defined(_MSC_VER) && defined(_M_X64))
                                             ^
poly1305-donna.c:18:51: error: macro expansion producing 'defined' has undefined behavior [-Werror,-Wexpansion-to-defined]
#if (HAS_SIZEOF_INT128_64BIT || HAS_MSVC_64BIT || HAS_GCC_4_4_64BIT)
                                                  ^
poly1305-donna.c:16:28: note: expanded from macro 'HAS_GCC_4_4_64BIT'
#define HAS_GCC_4_4_64BIT (defined(__GNUC__) && defined(__LP64__) && ((__GNUC__ > 4) || ((__GNUC__ == 4) && (__GNUC_MINOR__ >= 4))))
                           ^
poly1305-donna.c:18:51: error: macro expansion producing 'defined' has undefined behavior [-Werror,-Wexpansion-to-defined]
poly1305-donna.c:16:49: note: expanded from macro 'HAS_GCC_4_4_64BIT'
#define HAS_GCC_4_4_64BIT (defined(__GNUC__) && defined(__LP64__) && ((__GNUC__ > 4) || ((__GNUC__ == 4) && (__GNUC_MINOR__ >= 4))))
                                                ^
6 errors generated.
make[2]: *** [poly1305-donna.o] Error 1
make[1]: *** [poly1305-donna/poly1305-donna.a] Error 2
make[1]: *** Waiting for unfinished jobs....
```

* Linux + clang
```
poly1305-donna.c:18:6: error: macro expansion producing 'defined' has undefined behavior [-Werror,-Wexpansion-to-defined]
#if (HAS_SIZEOF_INT128_64BIT || HAS_MSVC_64BIT || HAS_GCC_4_4_64BIT)
     ^
poly1305-donna.c:14:34: note: expanded from macro 'HAS_SIZEOF_INT128_64BIT'
#define HAS_SIZEOF_INT128_64BIT (defined(__SIZEOF_INT128__) && defined(__LP64__))
                                 ^
poly1305-donna.c:18:6: error: macro expansion producing 'defined' has undefined behavior [-Werror,-Wexpansion-to-defined]
poly1305-donna.c:14:64: note: expanded from macro 'HAS_SIZEOF_INT128_64BIT'
#define HAS_SIZEOF_INT128_64BIT (defined(__SIZEOF_INT128__) && defined(__LP64__))
                                                               ^
poly1305-donna.c:18:33: error: macro expansion producing 'defined' has undefined behavior [-Werror,-Wexpansion-to-defined]
#if (HAS_SIZEOF_INT128_64BIT || HAS_MSVC_64BIT || HAS_GCC_4_4_64BIT)
                                ^
poly1305-donna.c:15:25: note: expanded from macro 'HAS_MSVC_64BIT'
#define HAS_MSVC_64BIT (defined(_MSC_VER) && defined(_M_X64))
                        ^
poly1305-donna.c:18:33: error: macro expansion producing 'defined' has undefined behavior [-Werror,-Wexpansion-to-defined]
poly1305-donna.c:15:46: note: expanded from macro 'HAS_MSVC_64BIT'
#define HAS_MSVC_64BIT (defined(_MSC_VER) && defined(_M_X64))
                                             ^
poly1305-donna.c:18:51: error: macro expansion producing 'defined' has undefined behavior [-Werror,-Wexpansion-to-defined]
#if (HAS_SIZEOF_INT128_64BIT || HAS_MSVC_64BIT || HAS_GCC_4_4_64BIT)
                                                  ^
poly1305-donna.c:16:28: note: expanded from macro 'HAS_GCC_4_4_64BIT'
#define HAS_GCC_4_4_64BIT (defined(__GNUC__) && defined(__LP64__) && ((__GNUC__ > 4) || ((__GNUC__ == 4) && (__GNUC_MINOR__ >= 4))))
                           ^
poly1305-donna.c:18:51: error: macro expansion producing 'defined' has undefined behavior [-Werror,-Wexpansion-to-defined]
poly1305-donna.c:16:49: note: expanded from macro 'HAS_GCC_4_4_64BIT'
#define HAS_GCC_4_4_64BIT (defined(__GNUC__) && defined(__LP64__) && ((__GNUC__ > 4) || ((__GNUC__ == 4) && (__GNUC_MINOR__ >= 4))))
                                                ^
6 errors generated.
Makefile:21: recipe for target 'poly1305-donna.o' failed
make[2]: *** [poly1305-donna.o] Error 1
Makefile:1435: recipe for target 'poly1305-donna/poly1305-donna.a' failed
make[1]: *** [poly1305-donna/poly1305-donna.a] Error 2
make[1]: *** Waiting for unfinished jobs....
ar: creating aes.a
ar: creating secp256k1.a
ar: creating ed25519-donna.a
Makefile:189: recipe for target 'default' failed
make: *** [default] Error 2
```
SIMD nothing better than AVX (?) (seems it is).
```
Testing: restic, Restic Repository [scrypt Salsa20/8 128/128 AVX, Poly1305]... (2xOMP) PASS
```
BSD
```
Fails (it is also clang)
```

--- a/src/poly1305-donna/poly1305-donna.c
+++ b/src/poly1305-donna/poly1305-donna.c
@@ -11,9 +11,23 @@
 #else
 
 /* auto detect between 32bit / 64bit */
-#define HAS_SIZEOF_INT128_64BIT (defined(__SIZEOF_INT128__) && defined(__LP64__))
-#define HAS_MSVC_64BIT (defined(_MSC_VER) && defined(_M_X64))
-#define HAS_GCC_4_4_64BIT (defined(__GNUC__) && defined(__LP64__) && ((__GNUC__ > 4) || ((__GNUC__ == 4) && (__GNUC_MINOR__ >= 4))))
+#if (defined(__SIZEOF_INT128__) && defined(__LP64__))
+#define HAS_SIZEOF_INT128_64BIT 1
+#else
+#define HAS_SIZEOF_INT128_64BIT 0
+#endif
+
+#if (defined(_MSC_VER) && defined(_M_X64))
+#define HAS_MSVC_64BIT 1
+#else
+#define HAS_MSVC_64BIT 0
+#endif
+
+#if (defined(__GNUC__) && defined(__LP64__) && ((__GNUC__ > 4) || ((__GNUC__ == 4) && (__GNUC_MINOR__ >= 4))))
+#define HAS_GCC_4_4_64BIT 1
+#else
+#define HAS_GCC_4_4_64BIT 0
+#endif
 
 #if (HAS_SIZEOF_INT128_64BIT || HAS_MSVC_64BIT || HAS_GCC_4_4_64BIT)
 #include "poly1305-donna-64.h"

